tests/page-objects/forms/generic-form.po.js

```diff
@@ -1,15 +1,36 @@
+const { element } = require('protractor');
 const helper = require('../../helper');
 const utils = require('../../utils');
 const nameField = element(by.css('#report-form form [name="/data/name"]'));
 const submitButton = element(by.css('.enketo .submit'));
 const submittedName = element(by.css('#reports-content .details ul li:first-child p'));
+const formTitle = element(by.id('form-title'));
+const nextButton = element(by.css('button.btn.btn-primary.next-page'));
 
 const leftActionBarButtons = () => element.all(by.css('.general-actions .actions.dropup > a'));
 
+const fillForm = async (formFields) => {
+  for (const [fieldName] of Object.entries(formFields)) {
+    const field = formFields[fieldName];
+    const elm = element(by.css(field.css));
+    if (field.textField) {
+      await elm.sendKeys(field.value);
+    } else {
+      const radioButton = element(by.css(`${field.css}[value=${field.value}]`));
+      await helper.clickElementNative(radioButton);
+    }
+    if (field.endOfPage){
+      await module.exports.nextPageNative();
```

Comment:
I worry we're mixing the concepts of "fill in the fields" with navigating between pages of a form. Ideally there is logic for filling in fields and separate logic for determining if a page needs to change. Then another function call call those two functions as needed.

Reply:
This is actually not supposed to be here. so good news it's going away!